src/network/networkd-sysctl.c

```diff
@@ -4,18 +4,209 @@
 #include <linux/if.h>
 #include <linux/if_arp.h>
 
+#include "sd-messages.h"
+
 #include "af-list.h"
+#include "cgroup-util.h"
+#include "fd-util.h"
 #include "missing_network.h"
 #include "networkd-link.h"
 #include "networkd-lldp-tx.h"
 #include "networkd-manager.h"
 #include "networkd-ndisc.h"
 #include "networkd-network.h"
 #include "networkd-sysctl.h"
+#include "path-util.h"
 #include "socket-util.h"
 #include "string-table.h"
 #include "sysctl-util.h"
 
+#if HAVE_VMLINUX_H
+
+#include "bpf-link.h"
+
+#include "bpf/sysctl_monitor/sysctl-monitor-skel.h"
+#include "bpf/sysctl_monitor/sysctl-write-event.h"
+
+static struct sysctl_monitor_bpf *sysctl_monitor_bpf_free(struct sysctl_monitor_bpf *obj) {
+        sysctl_monitor_bpf__destroy(obj);
+        return NULL;
+}
+
+static struct ring_buffer *rb_free(struct ring_buffer *rb) {
+        sym_ring_buffer__free(rb);
+        return NULL;
+}
+
+DEFINE_TRIVIAL_CLEANUP_FUNC(struct sysctl_monitor_bpf *, sysctl_monitor_bpf_free);
+DEFINE_TRIVIAL_CLEANUP_FUNC(struct ring_buffer *, rb_free);
+
+static int sysctl_event_handler(void *ctx, void *data, size_t data_sz) {
+        struct sysctl_write_event *we = ASSERT_PTR(data);
+        Hashmap **sysctl_shadow = ASSERT_PTR(ctx);
+        _cleanup_free_ char *path = NULL;
+        char *value;
+
+        /* Returning a negative value interrupts the ring buffer polling,
+         * so do it only in case of a fatal error like a version mismatch. */
+        if (we->version != 1)
+                return log_warning_errno(SYNTHETIC_ERRNO(EINVAL),
+                                "Unexpected sysctl event, disabling sysctl monitoring: %d", we->version);
+
+        if (we->errorcode < 0) {
+                log_warning_errno(we->errorcode, "Sysctl monitor BPF returned error: %m");
+                return 0;
+        }
+
+        path = path_join("/proc/sys", we->name);
+        if (!path) {
+                log_oom();
+                return 0;
+        }
+
+        /* If we never managed this handle, ignore it. */
+        value = hashmap_get(*sysctl_shadow, path);
+        if (!value)
+                return 0;
+
+        if (!strneq(value, we->newvalue, sizeof(we->newvalue)))
+                log_struct(LOG_WARNING,
+                           "MESSAGE_ID=" SD_MESSAGE_SYSCTL_CHANGED_STR,
+                           "COMM_PID=%d", we->pid,
+                           "COMM=%s", we->comm,
```

Comment:
can you name these OBJECT_PID= and OBJECT_COMM= please?

background: see the systemd.journald-fields man page: when we log *about* some other process we usually use that, so that we have a common key to search for, to find everything logged about some process.